crates/uv-resolver/src/candidate_selector.rs

```diff
@@ -78,6 +82,22 @@ impl CandidateSelector {
             }
         }
 
+        if !exclusions.contains(package_name) {
+            // Allow locally installed distributions to be candidates
```

Comment:
And prefer them over non-local packages, is my read. (But not over "preferences".)

Reply:
https://github.com/astral-sh/uv/pull/2596/commits/76fe2f43d5380e84a119af45df4dc6ba706cc500